core/description/resource.go

```diff
@@ -203,6 +203,70 @@ func (r *resourceRevision) Username() string {
 	return r.Username_
 }
 
+func (r *resourceRevision) merge(args ResourceRevisionArgs) (*resourceRevision, error) {
+	// Check fields match where set.
+	mkErr := func(field string) error {
+		return errors.Errorf("%s mismatch for revision %d", field, args.Revision)
+	}
+	checkStr := func(field, a, b string) error {
+		if a != "" && b != "" && a != b {
```

Comment:
No, it's ok for `b` to be empty. It won't get used in that case. The value could come from either side but if it's set on both sides then they have to match.